test/test_profiler.py

```diff
@@ -1184,28 +1193,71 @@ def end_time_ns(self):
+                MockKinetoEvent("GPU", 1000, 100, 2, 1),
+                MockKinetoEvent("GPU", 1100, 100, 3, 1),
+                MockKinetoEvent("GPU", 1200, 100, 4, 1),
+                MockKinetoEvent("GPU", 1300, 100, 5, 1),
+                MockKinetoEvent("GPU", 1700, 100, 6, 1)
+            ]
+        else:
+            json_list = json.loads(cuda_events_json)
+            cuda_events = [
+                MockKinetoEvent(*event.values()) for event in json_list
+            ]
 
-        cpu_events = [
-            MockProfilerEvent("CPU (Before cudaLaunchKernel)", 1, 0, 100000),
-            MockProfilerEvent("CPU (Before cudaLaunchKernel)", 2, 100001, 100000),
-            MockProfilerEvent("CPU (Before cudaLaunchKernel)", 3, 200001, 100000),
-            MockProfilerEvent("CPU (Before cudaLaunchKernel)", 4, 300001, 100000),
-            MockProfilerEvent("CPU (After cudaLaunchKernel)", 5, 400001, 100000),
-            MockProfilerEvent("CPU (After cudaLaunchKernel)", 6, 500001, 100000),
-            MockProfilerEvent("CPU (After cudaLaunchKernel)", 7, 600001, 100000),
-            MockProfilerEvent("CPU (After GPU)", 8, 700001, 100000),
-            MockProfilerEvent("CPU (After GPU)", 9, 800001, 100000),
-            MockProfilerEvent("CPU (After GPU)", 10, 900001, 100000),
-            MockProfilerEvent("CPU (No Event)", 11, 1000001, 100000),
-        ]
+        if cpu_events_json == "":
+            cpu_events = [
+                MockProfilerEvent("CPU (Before cudaLaunchKernel)", 1, 0,
+                                  100000),
+                MockProfilerEvent("CPU (Before cudaLaunchKernel)", 2, 100000,
+                                  100000),
+                MockProfilerEvent("CPU (Before cudaLaunchKernel)", 3, 200000,
+                                  100000),
+                MockProfilerEvent("CPU (Before cudaLaunchKernel)", 4, 300000,
+                                  100000),
+                MockProfilerEvent("CPU (After cudaLaunchKernel)", 5, 400000,
+                                  100000),
+                MockProfilerEvent("CPU (After cudaLaunchKernel)", 6, 500000,
+                                  100000),
+                MockProfilerEvent("CPU (After cudaLaunchKernel)", 7, 600000,
+                                  100000),
+                MockProfilerEvent("CPU (After cudaLaunchKernel)", 8, 700000,
+                                  100000),
+                MockProfilerEvent("CPU (After GPU)", 9, 800000, 100000),
+                MockProfilerEvent("CPU (After GPU)", 10, 900000, 100000),
+                MockProfilerEvent("CPU (After GPU)", 11, 1100000, 100000),
+                MockProfilerEvent("CPU (After GPU)", 12, 1200000, 500000),
+            ]
+        else:
+            json_list = json.loads(cpu_events_json)
+            cpu_events = []
+            id_map = {}
+            for e in json_list:
+                event = MockProfilerEvent(e['_name'],
```

Comment:
Is there any reason not to serialize these as a list so you can expand them as positional arguments like you did with `MockKinetoEvent`?